File: app/src/main/java/com/aero51/githubapp/retrofit/GithubApi.java
package com.aero51.githubapp.retrofit;

import com.aero51.githubapp.db.model.RepoSearchResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GithubApi {

    @GET("search/repositories?sort=stars")
    Call<RepoSearchResponse> searchRepos(@Query("q") String query,
                                         @Query("page") int page,
                                         @Query("per_page") int itemsPerPage);

}

File: app/src/main/java/com/aero51/githubapp/retrofit/GithubApi.kt
package com.aero51.githubapp.retrofit

import retrofit2.http.GET
import com.aero51.githubapp.model.RepoSearchResponse
import retrofit2.http.Query

interface GithubApi {
    //@GET("search/repositories?sort=stars")
    @GET("search/repositories")
    suspend fun searchRepos(@Query("q") query: String,
                            @Query("sort") sort: String,
                            @Query("page") page: Int,
                            @Query("per_page") itemsPerPage: Int): RepoSearchResponse
}